rust/operator-binary/src/zk_controller.rs

```diff
@@ -460,38 +493,49 @@ pub async fn reconcile_zk(
     }
 
     let role_config = zk.role_config(&zk_role);
-    if let Some(GenericRoleConfig {
-        pod_disruption_budget: pdb,
+    if let Some(ZookeeperServerRoleConfig {
+        common,
+        listener_class: _,
     }) = role_config
```

Comment:
Is there a reason to write it out?
```suggestion
    if let Some(ZookeeperServerRoleConfig { common, .. }) = role_config {
```